### ISSUE TITLE
Unable to set CORS rules programatically

### ISSUE THREAD

**Issue by GuyHarwood** [Azure/Azurite#55]
In order to push to a storage queue from another origin, i need to add the appropriate CORS rule.  At start up i try to set this programatically for the queue service....

```
  const properties = {
    Cors: {
      CorsRule: [{
        AllowedOrigins: ['*'],
        AllowedMethods: ['POST', 'GET', 'HEAD', 'PUT'],
        AllowedHeaders: ['*'],
        ExposedHeaders: ['*'],
        MaxAgeInSeconds: 3600
      }]
    }
  }

  qService.setServiceProperties(properties, function (error) {
    if (error) {
      throw error
    }
  })
```

This fails with the following error....

```
StorageError: Not Implemented yet.
    at Function.StorageServiceClient._normalizeError (~/dev/_spikes-poc/sas/server/node_modules/azure-storage/lib/common/services/storageserviceclient.js:1160:12)
    at QueueService.StorageServiceClient._processResponse (~/dev/_spikes-poc/sas/server/node_modules/azure-storage/lib/common/services/storageserviceclient.js:744:50)
    at Request.processResponseCallback [as _callback] (~/dev/_spikes-poc/sas/server/node_modules/azure-storage/lib/common/services/storageserviceclient.js:317:37)
    at Request.self.callback (~/dev/_spikes-poc/sas/server/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (~/dev/_spikes-poc/sas/server/node_modules/request/request.js:1157:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (~/dev/_spikes-poc/sas/server/node_modules/request/request.js:1079:12)
```

I also tried hitting the associated rest endpoints, but get 501 not implemented...

```
GET /devstoreaccount1?restype=service&comp=properties HTTP/1.1
501: Not Implemented

PUT /devstoreaccount1?restype=service&comp=properties HTTP/1.1
501: Not Implemented
```

Is there another convention for setting CORS rules in Azurite?

**Comment by jbfm** [comment]
Really missing CORS settings here as well. Difficult to test web implementations without it.

**Comment by XiaoningLiu** [comment]
Hi Guys, CORS support in Azurite V3 is planned in the incoming roadmap before V3 GA.

**Comment by kuba-lilz** [comment]
@XiaoningLiu Is there any timeline on cors settings release?

**Comment by XiaoningLiu** [comment]
@kuba-lilz Already released : ) Will close this issue.

**Comment by kuba-lilz** [comment]
@XiaoningLiu 
Hi, I just double checked using `mcr.microsoft.com/azure-storage/azurite:3.11.0` and no, CORS settings aren't reflected correctly, even though container accepts the request for setting them and returns 202.

I'm sending request with python azure-storage-blob==1.5.0 package and following code
```python
cors_rule = azure.storage.common.models.CorsRule(
        allowed_origins="0.0.0.0:*", 
        allowed_headers="*",
        allowed_methods=["GET"], 
        exposed_headers="*", 
        max_age_in_seconds=100)

blob_storage.set_blob_service_properties(cors=[cors_rule])
```

but when accessing data from a web browser I still get
```
edit:1 Access to image at 'http://localhost:10000/devstoreaccount1/photos/7d64ff9f-76ef-425e-81d1-e9478955b30e/0734507c-431a-4996-a1bd-597a1fb5708e.jpg?st=2021-03-10T07%3A47%3A48Z&se=2021-03-10T08%3A57%3A48Z&sp=r&sv=2018-03-28&sr=b&rsct=image/jpeg&sig=OwE3UF1dsgC30V80Xoenprhezd2wgKxzQY01sSFSRYk%3D' from origin 'http://0.0.0.0:5000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```
error

**Comment by swimhiking** [comment]
i try to manually set CORS, still not success, got 400 error, any suggestion to use azurite? thanks
![azurite-post-400error](https://user-images.githubusercontent.com/6053471/142130042-b76a0a4f-4343-4a36-99c5-184114403cb6.png)

**Comment by blueww** [comment]
@swimhiking 
You issue might not be same issue as before.
Would you please open a new issue by following the issue template (especially give the debug log of Azurite), and we can look deeper into it.
